blog/_posts/2024-12-04-mirrord-on-aks.md

```diff
@@ -0,0 +1,184 @@
+
+Tools like this (dubbed "remocal" development tools) are designed to improve your experience when developing applications intended to be deployed in a cluster. The point of mirrord is to shift left on cloud testing and shorten the dev loop from hours to seconds. It's the kind of thing you set up once and then can't imagine living without because it feels like magic.
+
+![Screenshot of the inner dev loop](/AKS/assets/images/mirrord-on-aks/mirrord-dev-loop.png)
+
+
+
+In short, mirrord lets you connect your local process with your development or staging cluster. It grabs env vars, files and traffic from the cluster in a way that makes your process think it's running remotely. The approach of mirrord is unique in that instead of simply starting a VPN, it hooks system calls in the local process and proxies traffic to the cluster.
+
+mirrord is also highly configurable, allowing for very granular specification of exactly what comes from the cluster and what comes from the local environment. Additionally, mirrord is easy to use with any kind of cluster configuration, no matter what service meshes or other more complex architecture it uses. It works just as well regardless of the size and shape of your cluster.
+
+Some other pros of using mirrord over similar solutions include:
+- no root privileges are required on your machine
+- super fast start up speed
+- you can run multiple local processes at the same time
+
+
+## Using mirrord with AKS
+
+mirrord is cloud agnostic and seamlessly supports AKS. Additionally, the setup process is exceedingly simple: as long as you can access your cluster with `kubectl`, you can run mirrord and start developing with a lightning fast feedback loop.
+
+Next up are a couple of walkthroughs for using the mirrord VSCode plugin to develop parts of the [AKS store demo sample](https://github.com/Azure-Samples/aks-store-demo); the [first example](#a-practical-example-rust) looks at `product-service`, a Rust application, and [the second](#another-practical-example-golang) looks at `makeline-service`, which is written in Go. Feel free to try both or whichever one interests you more. They are completely independent of each other, but I would recommend reading the Rust example even if you only want to follow the Go walkthrough.
+
+## A practical example: Rust
+
+And now for a guide on how to start using mirrord for yourself, using the [AKS store demo sample](https://github.com/Azure-Samples/aks-store-demo). You'll need the following:
+
+- An AKS cluster with the store demo deployed (you can use either the full deployment or the reduced one from `aks-store-quickstart.yaml` for this example), accessible from your machine with `kubectl`
+- Rust (I prefer to install the Rust compiler using [Rustup](https://rustup.rs/))
+- A local clone of the store demo
```

Comment:
maybe git clone command so people can just copy/paste and run instead of just listed as pre-requisite